server/src/room/routes.rs

```diff
@@ -0,0 +1,173 @@
+use crate::errors::ServiceError;
+use crate::room::{Room, RoomData};
+use crate::Pool;
+use actix_web::{delete, get, post, put, web, HttpResponse};
+use serde::Deserialize;
+use serde_json::json;
+use uuid::Uuid;
+
+#[get("/rooms")]
+pub async fn list(pool: web::Data<Pool>) -> Result<HttpResponse, ServiceError> {
+    let conn = pool.get().expect("couldn't get db connection from pool");
+
+    let rooms = web::block(move || Room::find_all(&conn))
+        .await
+        .map_err(ServiceError::from)?;
+
+    if rooms.is_empty() {
+        Err(ServiceError::NoContent)
+    } else {
+        Ok(HttpResponse::Ok().json(rooms))
+    }
```

Comment:
I found that `NoContent` is not good when an empty list is returned. https://stackoverflow.com/questions/13366730/proper-rest-response-for-empty-table

Furthermore I suggest returning a real json object and not only a list for extendability https://softwareengineering.stackexchange.com/questions/286293/whats-the-best-way-to-return-an-array-as-a-response-in-a-restful-api

```suggestion
    Ok(HttpResponse::Ok().json(json!({ "rooms": rooms })))
```